app/controllers/sessions_controller.rb

```diff
@@ -44,8 +48,12 @@ def create
       session[:login_state_id] = @login_state_id
 
       if request.env["warden.mfa.required"]
-        MultiFactorAuth.generate_and_send_code(resource)
-        redirect_to enter_phone_code_path
+        if webauthn_registered?(resource)
+          redirect_to webauthn_authenticate_path
+        else
+          MultiFactorAuth.generate_and_send_code(resource)
+          redirect_to enter_phone_code_path
+        end
       else
```

Comment:
I think we noticed that this doesn't work when MFA is skipped

Reply:
Yeah this needs more robust testing and a bit of service design I think.
Right at the end I think I managed to demo to Alan that it did skip MFA if we:
- started with no keys, logged in with SMS 2FA, checking remember me
- registered a key
- logged out
- logged back in, it went straight to account.

I think we'd want to consider if we add a similar remember me box to the webauth, or if that gets ignored for login / other auth points? And if so when  / where we should clear the cookie.

These fallbacks should probably also sit in some kind of preferred heirarchy (like if we have webauthn, do we always prefer to that SMS or OTP?) - then we would have looked to package up the Remember Me feature inside that new heirarchy? (what happens if you add a new factor... do we still remember you, or re-challenge?)